bundles/remote_services/topology_manager/src/topology_manager.c

```diff
@@ -288,6 +289,8 @@ celix_status_t topologyManager_rsaRemoved(void * handle, service_reference_pt re
 	remote_service_admin_service_t *rsa = (remote_service_admin_service_t *) service;
 
 	celixThreadMutex_lock(&manager->rsaListLock);
+	arrayList_removeElement(manager->rsaList, rsa);
+	celixThreadMutex_unlock(&manager->rsaListLock);
```

Comment:
`topologyManager_rsaRemoved` should be dual to `topologyManager_rsaAdded`: things should be done in strictly reverse order.

Reply:
Thank you for your valuable suggestions.
@pnoltes , If I understand your opinion, we can delete rsaListLock, exportedServicesLock and importedServicesLock , but  add a single mutex. Then we can  use the new  mutex protect rsaList,exportedServices, importedServices and relevant operation.

If I am right, I will also fix issue #415 in this PR.

Reply:
Yes correct. a single mutex in the topology manager should be enough.